### PR TITLE
Impl from trait for uint and bytes types

### PR DESCRIPTION
**Pull request by s1na**
Feel free to close the PR if it's not useful.

### CODE REVIEW

**Comment by axic** [comment]
What about this code: https://github.com/ewasm/ewasm-rust-api/pull/38/commits/dea1323fb8fdd6fb7896a0fa130deacbe5f04bcf ?

**Comment by jakelang** [comment]
@axic yes but those are just trait declarations for when we have no_std enabled.

File: src/types.rs [hunk 1]
```diff
@@ -42,3 +42,51 @@ pub type Hash = Bytes32;
...
+        Bytes32 { bytes: value }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Bytes20, Bytes32, Uint128, Uint256};
+
+    #[test]
+    fn test_from() {
```
Comment (axic) [inline src/types.rs]: Can you add `.into()` tests too?

**Comment by hugo-dc** [comment]
It is possible to add functionality in order to convert a StorageKey/StorageValue into an Address and the other way around?

**Comment by jakelang** [comment]
@hugo-dc Yes it would be possible via the type conversion traits. however I am not sure if i want to implicitly truncate the storage types to address types in the trait impl, as ideally there would be no data loss in those traits.

**Comment by axic** [comment]
> It is possible to add functionality in order to convert a StorageKey/StorageValue into an Address and the other way around?

I'm not fully sure it is needed because you could use these with slices.

**Comment by hugo-dc** [comment]
Here is some code I have for wrc20:

```rust
let input_data = ewasm_api::calldata_acquire();
let address_data = input_data[4..].to_vec();
let mut address = Address::default();
address.bytes.copy_from_slice(&address_data[0..20]);

let mut storage_key = StorageKey::default();
storage_key.bytes[12..].copy_from_slice(&address.bytes[0..20]);
```

I was wondering if it would be easier if the ewasm-rust-api lib could allow me to do something like this:

```rust
let input_data = ewasm_api::calldata_acquire();
let address = Address::from(&input_data[4..]);   // converting &[u8] -> Address

let storage_key = StorageKey::from(&address); // converting Address -> StorageKey
```

**Comment by axic** [comment]
@jakelang can you review please? @s1na has macroified it.

**Comment by s1na** [comment]
I was still thinking of implementing `From` smaller uints (and byte arrays) to wider ones (e.g. `Bytes20` -> `Bytes32`) and `TryFrom` for wider to smaller ones to make @hugo-dc 's example somewhat easier. Would that make sense?

**Comment by axic** [comment]
@jakelang happy to merge?

**Comment by jakelang** [comment]
looks good to merge.

**Pull request merged**